core/src/main/java/org/testcontainers/containers/JibDockerClient.java

```diff
@@ -0,0 +1,69 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectImageResponse;
+import com.google.cloud.tools.jib.api.DockerClient;
+import com.google.cloud.tools.jib.api.ImageDetails;
+import com.google.cloud.tools.jib.api.ImageReference;
+import com.google.cloud.tools.jib.http.NotifyingOutputStream;
+import com.google.cloud.tools.jib.image.ImageTarball;
+import com.google.common.io.ByteStreams;
+import com.google.common.io.CharStreams;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.images.RemoteDockerImage;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class JibDockerClient implements DockerClient {
+
+    private final com.github.dockerjava.api.DockerClient dockerClient = DockerClientFactory.lazyClient();
+
+    @Override
+    public boolean supported(Map<String, String> map) {
+        return true;
+    }
+
+    @Override
+    public String load(ImageTarball imageTarball, Consumer<Long> writtenByteCountListener) throws IOException {
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        imageTarball.writeTo(out);
+        ByteArrayInputStream in = new ByteArrayInputStream(out.toByteArray());
+        this.dockerClient.loadImageCmd(in).exec();
+        try (InputStreamReader stdout = new InputStreamReader(in, StandardCharsets.UTF_8)) {
+            return CharStreams.toString(stdout);
+        }
```

Comment:
Have you considered streaming operations here? `ImageTarball` could be pretty big and cause OOMs

Reply:
thanks for pointing out this, still looking at it.

Reply:
it has been implemented but depends on https://github.com/docker-java/docker-java/pull/1982